src/main/java/it/eng/idsa/dataapp/service/impl/ProxyServiceImpl.java

```diff
@@ -795,12 +796,23 @@ private ResponseEntity<String> handleResponse(ResponseEntity<String> resp, Multi
 			}
 			httpHeaders.put(HTTP.CONTENT_TYPE, List.of(payloadContentType));
 			httpHeaders.put(HTTP.CONTENT_LEN, List.of(String.valueOf(responsePayload.length())));
-			return ResponseEntity.status(HttpStatus.OK).headers(httpHeaders).body(responsePayload);
+
+			return ResponseEntity.status(HttpStatus.OK).headers(filterCorsHeaders(httpHeaders)).body(responsePayload);
 		}
 
 		return resp;
 	}
 
+	private HttpHeaders filterCorsHeaders(HttpHeaders headers) {
+		if (headers.containsKey("Access-Control-Allow-Origin")) {
```

Comment:
Do you need to check if headers.contains(key)?
What happens if calling remove directly, without check, will it fail if entry not present?

Reply:
Honestly, I see no reason for calling remove directly, without checking earlier if they exist, so I haven't tested the case if it fails when the entry is not present.